src/mcpbr/storage/cloud.py

```
@@ -0,0 +1,383 @@
+            local_path: Local file path to upload.
+            key: S3 object key (relative to prefix).
+
+        Returns:
+            Full S3 URI of uploaded object.
+
+        Raises:
+            CloudStorageError: If upload fails.
+        """
+        s3_uri = self._s3_path(key)
+        try:
+            subprocess.run(
```

Ruff (0.14.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)